### PR TITLE
Allow envvar values to be interpreted as JSON

### PR DESCRIPTION
**Pull request by jessemyers**

### CODE REVIEW

File: microcosm/loaders.py [hunk 1]
```diff
@@ -109,7 +109,10 @@ def matches_key(key_parts):
             dct[key_part.lower()] = dict()
             dct = dct[key_part.lower()]
         # set the value for the final part
-        dct[key_parts[-1].lower()] = value
+        try:
+            dct[key_parts[-1].lower()] = loads(value)
```
Comment (jessemyers) [inline microcosm/loaders.py]: This requires that the envvar value be quoted, which I think is an ok compromise.

```
FOO=bar
```
will be interpreted as a string

```
FOO="bar"
```

will be interpreted as a JSON string.

This makes it harder to have a literal quoted strings as values, but I'm not sure we need that.

The alternative is to have a different loader for JSON. Thoughts @adamhadani ?

File: microcosm/metadata.py [hunk 2]
```diff
@@ -23,7 +23,8 @@ def __init__(self, name, debug=False, testing=False, import_name=None, root_path
         self.name = name
         self.debug = debug
         self.testing = testing
-        self.root_path = root_path or self.get_root_path(import_name or name)
+        self.import_name = import_name or self.name
+        self.root_path = root_path or self.get_root_path(self.import_name)
```
Comment (jessemyers) [inline microcosm/metadata.py]: Unrelated change; I want this import value for Flask application initialization.

File: microcosm/loaders.py (hunk 1 above)
Comment (adamhadani) [inline microcosm/loaders.py]: @jessemyers not sure I fully understand let's talk IRL
